packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2382,23 +2578,39 @@ class UnconstrainedBox extends SingleChildRenderObjectWidget {
   /// Defaults to [Clip.none].
   final Clip clipBehavior;
 
+  BoxConstraintsTransform get _constraintsTransform {
+    final Axis? constrainedAxis = this.constrainedAxis;
+    if (constrainedAxis != null) {
+      switch (constrainedAxis) {
+        case Axis.horizontal:
+          return ConstraintsTransformBox._widthConstrained;
+        case Axis.vertical:
+          return ConstraintsTransformBox._heightConstrained;
+      }
+    } else {
+      return ConstraintsTransformBox._unconstrained;
+    }
+  }
+
+  @override
+  RenderConstraintsTransformBox createRenderObject(BuildContext context) {
```

Comment:
This could also just be implemented as a stateless widget now which builds a ConstraintsTransformBox?